optaplanner-examples/src/main/java/org/optaplanner/examples/examination/solver/score/ExaminationConstraintProvider.java

```diff
@@ -0,0 +1,241 @@
+                        filtering((roomPenalty, leftExam) -> leftExam.getPeriod() != null && leftExam.getRoom() != null))
+                .join(Exam.class,
+                        equal((roomPenalty, leftExam) -> leftExam.getRoom(), Exam::getRoom),
+                        equal((roomPenalty, leftExam) -> leftExam.getPeriod(), Exam::getPeriod),
+                        filtering((roomPenalty, leftExam, rightExam) -> leftExam.getTopic() != rightExam.getTopic()))
+                .penalize("Room penalty exclusive", HardSoftScore.ONE_HARD,
+                        (periodPenalty, leftExam, rightExam) -> leftExam.getTopic().getStudentSize()
+                                + rightExam.getTopic().getStudentSize());
+    }
+
+    protected Constraint twoExamsInARow(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(TopicConflict.class)
+                .join(Exam.class,
+                        equal(TopicConflict::getLeftTopic, Exam::getTopic),
+                        filtering((topicConflict, leftExam) -> leftExam.getPeriod() != null))
+                .ifExists(Exam.class,
+                        equal((topicConflict, leftExam) -> topicConflict.getRightTopic(), Exam::getTopic),
+                        equal((topicConflict, leftExam) -> leftExam.getDayIndex(), Exam::getDayIndex),
+                        filtering((topicConflict, leftExam,
+                                rightExam) -> getPeriodIndexDifferenceBetweenExams(leftExam, rightExam) == 1))
+                .join(ExaminationConstraintConfiguration.class)
+                .penalize("Two exams in a row", HardSoftScore.ONE_SOFT,
+                        (topicConflict, leftExam, config) -> topicConflict.getStudentSize() * config.getTwoInARowPenalty());
+    }
+
+    protected Constraint twoExamsInADay(ConstraintFactory constraintFactory) {
+        return constraintFactory.from(TopicConflict.class)
+                .join(Exam.class,
+                        equal(TopicConflict::getLeftTopic, Exam::getTopic),
+                        filtering((topicConflict, leftExam) -> leftExam.getPeriod() != null))
+                .ifExists(Exam.class,
+                        equal((topicConflict, leftExam) -> topicConflict.getRightTopic(), Exam::getTopic),
+                        equal((topicConflict, leftExam) -> leftExam.getDayIndex(), Exam::getDayIndex),
+                        // Find exams in a day, but not being held right after each other. That case is handled in the twoExamsInARow constraint.
+                        filtering((topicConflict, leftExam,
+                                rightExam) -> getPeriodIndexDifferenceBetweenExams(leftExam, rightExam) > 1))
+                .join(ExaminationConstraintConfiguration.class)
+                .penalize("Two exams in a day",
+                        HardSoftScore.ONE_SOFT,
+                        (topicConflict, leftExam, config) -> topicConflict.getStudentSize() * config.getTwoInADayPenalty());
```

Comment:
I see that there is a `ConstraintConfiguration` available. Yet, you still use `penalize()` instead of `penalizeConfigurable()`. I suggest you change that.